OSDOCS10847: Added placementGroupPartition field in machine set AWS existing placement group (#81645)
openshift/openshift-docs: modules/machineset-aws-existing-placement-group.adoc

Reviewer: I believe this is the only module where we've docs for `placementGroupName`.

Author: Yeah. And will update in the release notes as well.

            availabilityZone: <zone> # <3>
            region: <region> # <4>
          placementGroupName: <placement_group> # <5>

>           placementGroupPartition: <placement partition number> # <6>

Reviewer: ```suggestion
          placementGroupPartition: <placement_group_partition_number> # <6>
```

Reviewer: @subhtk could you pls update it as well? Thanks!

# ...
----
<1> Specify an instance type that link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html#efa-instance-types[supports EFAs].
<2> Specify the `EFA` network interface type.
<3> Specify the zone, for example, `us-east-1a`.
<4> Specify the region, for example, `us-east-1`.
<5> Specify the name of the existing AWS placement group to deploy machines in.

> <6> Specify the partition number of the existing AWS placement group to deploys machines in. 

Reviewer: Can we somehow write that this field is optional?

> .Verification

Reviewer: One line can be added to the `Verification` for `placementGroupPartition` like `placementGroupName`

            availabilityZone: <zone> # <3>
            region: <region> # <4>
          placementGroupName: <placement_group> # <5>
          placementGroupPartition: <placement_group_partition_number> # <6>
# ...
----
<1> Specify an instance type that link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html#efa-instance-types[supports EFAs].
<2> Specify the `EFA` network interface type.
<3> Specify the zone, for example, `us-east-1a`.
<4> Specify the region, for example, `us-east-1`.
<5> Specify the name of the existing AWS placement group to deploy machines in.

> <6> Optional: Specify the partition number of the existing AWS placement group to deploys machines in. 

Reviewer 2: ```suggestion
<6> Optional: Specify the partition number of the existing AWS placement group to deploy machines in. 
```

.Verification

* In the AWS console, find a machine that the machine set created and verify the following in the machine properties:

** The placement group field has the value that you specified for the `placementGroupName` parameter in the machine set.

> ** The placement group partition field has the value that you specified for the `placementGroupPartition` parameter in the machine set.

Reviewer: nit: checked in the AWS console; the field name is `Partition number` 
```suggestion
** The partition number field has the value that you specified for the `placementGroupPartition` parameter in the machine set.
```

            availabilityZone: <zone> # <3>
            region: <region> # <4>
          placementGroupName: <placement_group> # <5>
          placementGroupPartition: <placement partition number> # <6>
# ...
----
<1> Specify an instance type that link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html#efa-instance-types[supports EFAs].
<2> Specify the `EFA` network interface type.
<3> Specify the zone, for example, `us-east-1a`.
<4> Specify the region, for example, `us-east-1`.
<5> Specify the name of the existing AWS placement group to deploy machines in.
[Optional]

> <6> Specify the partition number of the existing AWS placement group to deploys machines in. 

Reviewer 3: Not sure if it needs to be stated that partition numbers can only be used on placement groups using the "partition" strategy

Reviewer: It's good to add it explicitly, but in that case, there are other constraints as well, like partition number - which we would need to add. 

I see there is a note just above the spec, with the link to the AWS documentation. I think this should suffice with everything related to pre-requisite to use this field. WDYT?
```
Ensure that the link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/placement-groups.html#limitations-placement-groups[rules and limitations] for the type of placement group that you create are compatible with your intended use case.
```

Reviewer 3: Yeah, that should be good.

Reviewer 3: LGTM

Author: Great
